src/adapter/graphql/src/mutations/datasets_mut.rs

```diff
@@ -94,23 +106,40 @@ impl DatasetsMut {
             }
         };
 
-        self.create_from_snapshot_impl(ctx, snapshot).await
+        self.create_from_snapshot_impl(
+            ctx,
+            snapshot,
+            dataset_publicly_available.unwrap_or_default(),
+        )
+        .await
     }
 
     // TODO: Multi-tenancy
     // TODO: Multi-tenant resolution for derivative dataset inputs (should it only
-    // work by ID?)
+    //       work by ID?)
     #[allow(unused_variables)]
     #[graphql(skip)]
     async fn create_from_snapshot_impl(
         &self,
         ctx: &Context<'_>,
         snapshot: odf::DatasetSnapshot,
+        dataset_visibility: bool,
     ) -> Result<CreateDatasetFromSnapshotResult> {
         let create_from_snapshot =
             from_catalog::<dyn domain::CreateDatasetFromSnapshotUseCase>(ctx).unwrap();
 
-        let result = match create_from_snapshot.execute(snapshot).await {
+        let create_options = CreateDatasetFromSnapshotUseCaseOptions {
+            dataset_visibility: if dataset_visibility {
+                DatasetVisibility::PubliclyAvailable
+            } else {
+                DatasetVisibility::Private
+            },
```

Comment:
Minor: might look nicer if encapsulated within the enum itself

Reply:
Yes, indeed -- there's no point in having a flag as an argument if we already have enumeration

Fixed